src/Analysis/Ast/Impl/Types/PythonClassType.Generics.cs

```diff
@@ -67,10 +68,10 @@ internal partial class PythonClassType {
                 // type parameter T -> int, U -> str, etc.
                 var genericTypeToSpecificType = GetSpecificTypes(args, genericTypeParameters, newBases);
 
-                PythonClassType classType = new PythonClassType(BaseName, new Location(DeclaringModule));
+                var classType = new PythonClassType(BaseName, new Location(DeclaringModule));
                 // Storing generic parameters allows methods returning generic types 
                 // to know what type parameter returns what specific type
-                StoreGenericParameters(classType, genericTypeParameters, genericTypeToSpecificType);
+                classType.StoreGenericParameters(genericTypeParameters, genericTypeToSpecificType);
```

Comment:
Doesn't changing this mean that `StoreGenericParameters` is now accessing the wrong class's `GenericParameters` property? Inside, it uses `GenericParameters` to copy over the original parameters, which before where the current class', but I think it'll be empty now that it's the new type's.

Reply:
No, it effectively used `classType` as `this`